src/Hamiltonians/ExtendedHubbardReal1D.jl

```diff
@@ -64,15 +64,19 @@ function LOStructure(::Type{<:ExtendedHubbardReal1D{<:Real,<:Any,<:Any,<:Any,T}}
     end
 end
 function LOStructure(::Type{<:ExtendedHubbardReal1D{<:Complex,<:Any,U,V,T}}) where {U,V,T}
-    if iszero(T)
-        return IsDiagonal() # TODO: implement adjoint
-    elseif iszero(imag(U)) && iszero(imag(V))
-        return IsHermitian() # still Hermitian with complex t and twisted boundaries
+    if iszero(imag(U)) && iszero(imag(V))
+        return IsHermitian() # still Hermitian with complex t
     else
-        return AdjointUnknown() # diagonal elements are complex; TODO: implement adjoint
+        return AdjointKnown()
     end
 end
 
+function LinearAlgebra.adjoint(h::ExtendedHubbardReal1D{TT,A,U,V,T,B}) where {TT<:Complex,A,U,V,T,B}
+    CU = imag(U) == 0 ? U : real(U) - im*imag(U)
+    CV = imag(V) == 0 ? V : real(V) - im*imag(V)
+    return ExtendedHubbardReal1D{TT,A,CU,CV,T,B}(h.address)
```

Comment:
Why not this way?
```suggestion
    return ExtendedHubbardReal1D{TT,A,conj(U),conj(V),T,B}(h.address)
```
It is more concise and more readable, no?
The function `conj` is a no-op if the argument is a real number, so you don't save anything by the `if` condition. Moreover, the whole function evaluation should be handled by the compiler, as it operates on type information only, so it has no runtime cost at all. Benchmarking seems to show that as well:
```julia-repl
julia> h = HubbardReal1D(BoseFS(1,0,3,0); t=1.0+im, u = 1.0+2im)
HubbardReal1D(fs"|1 0 3 0⟩"; u=1.0 + 2.0im, t=1.0 + 1.0im)

julia> using Chairmarks

julia> @b get_offdiagonal($h, $(starting_address(h)), 3)
14.949 ns

julia> @b get_offdiagonal(adjoint($h), $(starting_address(h)), 3)
14.952 ns

julia> @b diagonal_element($h, $(starting_address(h)))
4.154 ns

julia> @b diagonal_element(adjoint($h), $(starting_address(h)))
4.150 ns
```
with either way of defining the function.

Reply:
Yes this does work, I just ran into issues since `conj(1.0 + 0.0im) = 1.0 - 0.0im`, so for example with complex `t` and real `u`, the Hamiltonian is Hermitian but the adjoint was different and some tests failed.

Reply:
Hmm, interesting.
Indeed, 
```julia-repl
julia> z = complex(3.0)
3.0 + 0.0im

julia> z'
3.0 - 0.0im

julia> z' == z
true

julia> bitstring(z.im)
"0000000000000000000000000000000000000000000000000000000000000000"

julia> bitstring(z'.im)
"1000000000000000000000000000000000000000000000000000000000000000"
```
So when comparing the numbers they are treated as equal, but their bit representations are not.

Reply:
https://en.wikipedia.org/wiki/Signed_zero#Arithmetic

```julia
return ExtendedHubbardReal1D{TT,A,conj(U)+0im,conj(V)+0im,T,B}(h.address)
```
should do the job and remove the negative zeros.

Reply:
Yes that's a lot better, thanks.